### ISSUE TITLE
Compiler: Hard crash when deconstructing single-valued tuple

### ISSUE THREAD

**Issue by SLaks** [dotnet/roslyn#16068]
**Version Used**: 	a2df65047264ed08fb3a985056085dbcee821253

```C#
var (x) = ValueTuple.Create(1);
```

The compiler crashes [here](http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp/Symbols/Tuples/TupleTypeSymbol.cs,87) because single-valued tuples aren't supposed to exist.

Even worse, typing the last `)` into VS will CTD the entire IDE 💥 
```
	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.TupleTypeSymbol.Create(Microsoft.CodeAnalysis.Location locationOpt, System.Collections.Immutable.ImmutableArray<Microsoft.CodeAnalysis.CSharp.Symbols.TypeSymbol> elementTypes, System.Collections.Immutable.ImmutableArray<Microsoft.CodeAnalysis.Location> elementLocations, System.Collections.Immutable.ImmutableArray<string> elementNames, Microsoft.CodeAnalysis.CSharp.CSharpCompilation compilation, Microsoft.CodeAnalysis.CSharp.CSharpSyntaxNode syntax, Microsoft.CodeAnalysis.DiagnosticBag diagnostics)	Unknown
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Binder.MakeDeconstructionConstructionStep(Microsoft.CodeAnalysis.CSharp.CSharpSyntaxNode node, Microsoft.CodeAnalysis.DiagnosticBag diagnostics, System.Collections.Immutable.ImmutableArray<Microsoft.CodeAnalysis.CSharp.BoundDeconstructValuePlaceholder> constructionInputs)	Unknown
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Binder.DeconstructOrAssignOutputs(Microsoft.CodeAnalysis.CSharp.BoundDeconstructionDeconstructStep deconstructionStep, Microsoft.CodeAnalysis.ArrayBuilder<Microsoft.CodeAnalysis.CSharp.Binder.DeconstructionVariable> variables, Microsoft.CodeAnalysis.CSharp.CSharpSyntaxNode syntax, Microsoft.CodeAnalysis.DiagnosticBag diagnostics, Microsoft.CodeAnalysis.ArrayBuilder<Microsoft.CodeAnalysis.CSharp.BoundDeconstructionDeconstructStep> deconstructionSteps, Microsoft.CodeAnalysis.ArrayBuilder<Microsoft.CodeAnalysis.CSharp.BoundDeconstructionAssignmentStep> conversionSteps, Microsoft.CodeAnalysis.ArrayBuilder<Microsoft.CodeAnalysis.CSharp.BoundDeconstructionAssignmentStep> assignmentSteps, Microsoft.CodeAnalysis.ArrayBuilder<Microsoft.CodeAnalysis.CSharp.BoundDeconstructionConstructionStep> constructionStepsOpt)	Unknown
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Binder.DeconstructIntoSteps(Microsoft.CodeAnalysis.CSharp.BoundDeconstructValuePlaceholder targetPlaceholder, Microsoft.CodeAnalysis.CSharp.CSharpSyntaxNode syntax, Microsoft.CodeAnalysis.DiagnosticBag diagnostics, Microsoft.CodeAnalysis.ArrayBuilder<Microsoft.CodeAnalysis.CSharp.Binder.DeconstructionVariable> variables, Microsoft.CodeAnalysis.ArrayBuilder<Microsoft.CodeAnalysis.CSharp.BoundDeconstructionDeconstructStep> deconstructionSteps, Microsoft.CodeAnalysis.ArrayBuilder<Microsoft.CodeAnalysis.CSharp.BoundDeconstructionAssignmentStep> conversionSteps, Microsoft.CodeAnalysis.ArrayBuilder<Microsoft.CodeAnalysis.CSharp.BoundDeconstructionAssignmentStep> assignmentSteps, Microsoft.CodeAnalysis.ArrayBuilder<Microsoft.CodeAnalysis.CSharp.BoundDeconstructionConstructionStep> constructionStepsOpt)	Unknown
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Binder.BindDeconstructionAssignment(Microsoft.CodeAnalysis.CSharp.CSharpSyntaxNode node, Microsoft.CodeAnalysis.CSharp.Syntax.ExpressionSyntax right, Microsoft.CodeAnalysis.ArrayBuilder<Microsoft.CodeAnalysis.CSharp.Binder.DeconstructionVariable> checkedVariables, Microsoft.CodeAnalysis.DiagnosticBag diagnostics, Microsoft.CodeAnalysis.CSharp.BoundDeconstructValuePlaceholder rhsPlaceholder)	Unknown
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Binder.BindDeconstruction(Microsoft.CodeAnalysis.CSharp.Syntax.AssignmentExpressionSyntax node, Microsoft.CodeAnalysis.DiagnosticBag diagnostics)	Unknown
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Binder.BindAssignment(Microsoft.CodeAnalysis.CSharp.Syntax.AssignmentExpressionSyntax node, Microsoft.CodeAnalysis.DiagnosticBag diagnostics)	Unknown
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Binder.BindExpressionInternal(Microsoft.CodeAnalysis.CSharp.Syntax.ExpressionSyntax node, Microsoft.CodeAnalysis.DiagnosticBag diagnostics, bool invoked, bool indexed)	Unknown
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Binder.BindExpression(Microsoft.CodeAnalysis.CSharp.Syntax.ExpressionSyntax node, Microsoft.CodeAnalysis.DiagnosticBag diagnostics, bool invoked, bool indexed)	Unknown
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Binder.BindExpressionStatement(Microsoft.CodeAnalysis.CSharp.CSharpSyntaxNode node, Microsoft.CodeAnalysis.CSharp.Syntax.ExpressionSyntax syntax, bool allowsAnyExpression, Microsoft.CodeAnalysis.DiagnosticBag diagnostics)	Unknown
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Binder.BindExpressionStatement(Microsoft.CodeAnalysis.CSharp.Syntax.ExpressionStatementSyntax node, Microsoft.CodeAnalysis.DiagnosticBag diagnostics)	Unknown
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Binder.BindStatement(Microsoft.CodeAnalysis.CSharp.Syntax.StatementSyntax node, Microsoft.CodeAnalysis.DiagnosticBag diagnostics)	Unknown
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.MemberSemanticModel.IncrementalBinder.BindStatement(Microsoft.CodeAnalysis.CSharp.Syntax.StatementSyntax node, Microsoft.CodeAnalysis.DiagnosticBag diagnostics)	Unknown
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.CSharpSemanticModel.Bind(Microsoft.CodeAnalysis.CSharp.Binder binder, Microsoft.CodeAnalysis.CSharp.CSharpSyntaxNode node, Microsoft.CodeAnalysis.DiagnosticBag diagnostics)	Unknown
```

http://tryroslyn.azurewebsites.net/#b:master/f:r/K4Zwlgdg5gBAygTxAFwKYFsDcAoUlaIoYB0AwgPYA2lqAxsmORCDntPEmumVTfY82IBxVBFQAnMLVbh2hLsQAiYAIZQI5FFJa5ZBTiQBywdBO3EAEipAALfDPwci3AErAIDU8QCSHieQAHOAkANylUHWwA4AAjSikYWkprEBhSGABvbABIaLiEkPIwABMYAFkACgBKTOwYeob6kJVxGAqADxqAXhgANRVKYFQAFWAAmjJxVBU0CoBGKpxsgF9sVYgVUxAAlVpUJy5anMg0cQ3KGBOJADNd/e9R8dRfU/Pa7OyTmBFkK1sKYqoCreACiAEdgAMwMgEBR0Dspq1aOR4S0JIscp8PPAwAAvfYZGBQVDITAwVYfFCSdjDchwZDUqAgiDFapLVY5PLxWgwKnAeh9AZDR40GAALhgoIhMxUcVQAB5+oMRmMaAA+AA0kvp4n5yGA4gG4MhyFlNC13h1eoNAzhCLNqAtdpaDqdKPtcsVQpVT01kpFzz8ZwG71ysW5MHIIQkkkBMBi5CoMGNAxAFXIMQAVnRkJGszUsh9slN9Wc85nLqklcLVaglitsJiuQkE0mU5Q09WfaLyMgbOjQ8WSQaIDAGUN6xyPq2Lpbx/QbZQU6a5cR22mM9mBb3++ILe3obD3WikcfEQXMUPS6OdxJK4LlQHJ42Pl9vM7DauPyNyOmsznIz7AdC1fa42lvVouh6CBgGoGoSxHGA5hwS8ELLAAGZ8m3DBIvm/WkKi7ANAN3C8izQ0dMMxKcsVzOddQXQ1KA/B0eFRKYCM3ACILddi72RPjxDI0DwKAyDoNgyh4OHMtkMbVCZMorCPmbHkoxjEp9i+H4/hsAEgWEq9EKoj4aLfK1GKNaUVwmHTrD08hAWBA8YW/U9BMMiiYBMhtMTfAMXgkc5hBJXT9Oc6V4lcs9+JioTBy8nyaNUyNo3EWN9ipRxaR1fBqgSxSYAAImqIrlOyLL2AeWtAuDShiByhl8GZVlPMKoqqjK6iX1o/0aqDYK4DxfYujVbyUJUnCeRQGYEiI2s0imGYDJgUbsDEAB3B8ayeNkepSmaGB5eannlYY5jG0glrQM6Loqc7Li4BZVrVdbUC2k6aFutUKmhDAFgmsN8mm00ju27sFXOrVhgAJku67IbmaG4fuuZHv+5H0fQGHulezbwYDW7kZ+v70CRrGccBq46t5UHcOxOEYkgVAwsciJfuxGxya+GwcYK69xu6nIpwO+dc0+xHXolKUTQdL1H1rb6/Xo60mOXV1tTFxcWLlXiPXNSUda+iWlYtAKBpDECUoe7wnsBlKJdRrHnpA7Jbf+1bnfKlL1PSzT40TC51z/Ldc03NqBc3e8TfOsaADI4+TSK00I71Cdjqpw+9qaA7bZPU4V07Y5I4CFIFlyj0E76lFQW5YOQNd8/dsmtQgnwnoxUyepnTWGNLKzZdXYOuO3MT90iw83MSOLDLAMD0zE1aJOoGAAB9V5gABCBfd2jtPFYz6SBduDs6x67JmlaTdiJ6Audq+jOIMBoyywEhEJEbyEO2BJ7W6zAN27/U7r5V82J3xnlYvhX8I8w5iVnvPCCS8YAwTgjALyclLxzzaNvRBYAqz7yLhdKohlsh9nEOQLa+MACC4goAmFEMgEE7Q9gBAYEwCoRUipag2KYcg1wd7omASAj4l9yw3zaDHIhT9z5eTctXRQtcVD1zYu/H+GMxG1kAWTYByUc54RigRSRY0IIR0QnI2ONc66UAbt+NRLcS7iC0QDIWoC6IWX7sxCBX4DHQP/KPXcesTzTw8oOLBAjxLIMkkfRCGCixhJwYvPBBMD5EJIWQihyD3owBoXQ0wHgmEsLYRADhXDkGbFQHw8JxD6zCIvi0DRTxPZ3wht9TOYln5eTfieFRaI7Hk2vpo5uziu7YWBqlDScZtKhXsuFUxZYK7mIupYpR1iQq/BmWzPpOiermS1mrayrE7L/E2TLKE0VBLBPfvFV2nS4prNZk5IZ2y/JgPNq8AY9yNmPIWXFS5J45mji6YiT5xzHkd2zmM32GVaaMhgI1Rk+UbntQ4TAAA1JKJ6AB+BqdImrQHyuijqXURmUjxbAaqTxarBXhc1FkiKy6ISGdiml+KaiEs6uVfy/V3n1SGviF6SF2QvgANruIADIqAQOQYAyAKgSqlTKgA0pAYoxAqEyvIFUAAupyHOfIBSSOJtgaW6tPSGrhXDZW7jFymoNirSynj2Ia3AU63WhsvHGwIQ/cmsM1TKzeUFS2oyIw2ztvtHOsMMUYBhvbHOjsHqkx9TDCmhkhme0TTU5uyaeikxjS4oGEYoX+x7sPPxYd8z80QlHJJ5rfUwATknL+Kda1w0zvmCFEYS35xbcYuBlb5kT3OVc+RijlHByGa3MSTiqiXmyA2n5VdfXLLHU3LgMNJ27i0ZTfNPd7UeNtagT+qYQ7cTHpKBdVy/nnlCQgxeUFIkr3XlvcJe9C7euJsQtBhUT4gDPpeURAzGm3x7W03cHTCpAo/uO3+DSJhPNnQ2yDjjoPRr/pmABWbnmuPda6iYUCT3+NLnE29u970oKkl+gWsSRIVASbvGtXrEYfrSTYchlCsk5Pofk5hqBWECBKdw8plSTFCJoi/Uc+HmnpyTWqUD6IO301zPhntDjZ7Yh5D0RZapl2rNsROhx06alhJ5JvHo6Fomv3AwLcxcMdM2JinY9dBnMOcrAdapiRtD34ZgQ4wJiIr1EZo4gsjUTKMxOfvEl9DG31MYtbJljbHMlbU43kxhPG+PsM4YJ3h/CRM1LEwB/+C1gOMaJnFuT4gIvqc9khnpUw+loYw+CzB88TNmYs4CqziFat6bXY1wZa6sMFoSEWyZ2IjkOScgC5JTw2JMzEA89mF6TwjqsQ3Cb4UnkalnUWXbDRluIjOrZhRa2QWTaBJhobuy+42oOauDbJyDuxRCUigWD3AQUHq0hobARJDNDQJccb0zQWoE+xdp77krnTYlnN5mi20y1fexdju23dto4+A0RHwPzuOeqfmrllKLb1SRxFL+k9fnff7aOJHYOKjfYU2pNK0LKqwBZVAel5FkVFTRVGsmzLcUIrZcVLU3P0VZv57lVlPOiXlRZ31QnPKcWS6ZHS6bTKldkoJcL4qPPxca8F9Ljl+PXncsDby4aAq80NmWEAA===

**Comment by 0xd4d** [comment]
#15893

**Comment by jcouv** [comment]
This was fixed by #15922 which will ship as part of RC.3.
An error is now reported, as we do not allow 1-deconstruction at this point (why deconstruct into parts if there is only one part?).
